### PR TITLE
[PUI] Make Category tree theme-aware

### LINKED ISSUE #5824: [PUI] Tree side panel does not adhere to theme
image

Needs to use the selected theme / style

### PR DESCRIPTION
**Pull request by LavissaWoW**
Closes #5824

### CODE REVIEW

**Review by matmair — APPROVED** [review]
LGTM, thank you

**Comment by SchrodingersGat** [comment]
Great work @LavissaWoW :)

File: src/frontend/src/components/nav/PartCategoryTree.tsx [hunk 1]
```diff
@@ -59,6 +66,56 @@ export function PartCategoryTree({
     );
   }
 
+  const mantineTheme = useMantineTheme();
+  const currentTheme =
+    mantineTheme.colorScheme === 'dark'
+      ? mantineTheme.colorScheme
+      : mantineTheme.primaryColor;
+
+  const themes: ThemeSettings = {
```
Comment (wolflu05) [inline src/frontend/src/components/nav/PartCategoryTree.tsx:75]: It would make sense to wrap this into a useMemo for performance reasons.

Comment (LavissaWoW) [reply src/frontend/src/components/nav/PartCategoryTree.tsx:75]: What would the dependency be then? Output of `useMantineTheme()`?

Comment (wolflu05) [reply src/frontend/src/components/nav/PartCategoryTree.tsx:75]: Yes, every variable that is used inside the useMemo

**Pull request merged**